specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json

```diff
@@ -0,0 +1,2118 @@
+          "type": "string"
+        },
+        "expiresIn": {
+          "format": "int32",
+          "description": "Time in seconds that the token remains valid",
+          "type": "integer"
+        }
+      }
+    },
+    "BuildTaskFilter": {
+      "description": "The filter that can be used for listing build tasks.",
+      "type": "object",
+      "properties": {
+        "alias": {
+          "description": "The alternative name for build task.",
+          "type": "string"
+        }
+      }
+    },
+    "BuildTaskUpdateParameters": {
+      "description": "The parameters for updating a build task.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/BuildTaskPropertiesUpdateParameters",
+          "description": "The properties for updating a build task.",
+          "x-ms-client-flatten": true
+        },
+        "tags": {
+          "description": "The ARM resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "BuildTaskPropertiesUpdateParameters": {
+      "description": "The properties for updating a build task.",
+      "type": "object",
+      "properties": {
+        "alias": {
+          "description": "The alternative updatable name for a build task.",
+          "type": "string"
+        },
+        "status": {
+          "description": "The current status of build task.",
+          "enum": [
+            "Disabled",
+            "Enabled"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "BuildTaskStatus",
+            "modelAsString": true
+          }
+        },
+        "platform": {
+          "$ref": "#/definitions/PlatformProperties",
+          "description": "The platform properties against which the build has to happen."
+        },
+        "timeout": {
+          "format": "int32",
+          "description": "Build timeout in seconds.",
+          "type": "integer"
+        },
+        "sourceRepository": {
+          "$ref": "#/definitions/SourceRepositoryUpdateParameters",
+          "description": "The properties that describes the source(code) for the build task."
+        }
+      }
+    },
+    "SourceRepositoryUpdateParameters": {
+      "description": "The properties for updating the source code repository configuration.",
+      "type": "object",
+      "properties": {
+        "sourceControlAuthProperties": {
+          "$ref": "#/definitions/SourceControlAuthInfo",
+          "description": "The authorization properties for accessing the source code repository."
+        },
+        "isCommitTriggerEnabled": {
+          "description": "The value of this property indicates whether the source control commit trigger is enabled or not.",
+          "default": false,
+          "type": "boolean"
+        }
+      }
+    },
+    "QueueBuildRequest": {
```

Comment:
>QueueBuildRequest [](start = 5, length = 17)

Not the biggest fan of this definition name

Reply:
This was named after the operation name QueueBuild and it is polymorphic type and its subtype makes more sense like QuickBuild and BuildTask request.